deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java

```diff
@@ -230,16 +235,71 @@ private Optional<String> getInputBaseDirRelativeToModule(final Path sourceDir, f
             Class<K> kClass, Class<V> vClass) {
+
+        Optional<String> possibleConfigKey = getConfigKeyValue(config, openApiFilePath);
+        boolean userConfiguredConfigKey = possibleConfigKey.isPresent() && !possibleConfigKey.get().isEmpty();
+
+        if (userConfiguredConfigKey) {
+            return getValuesByConfigKey(config, configName, kClass, vClass, possibleConfigKey.get());
+        }
+
+        return getValuesBySpecConfigName(config, openApiFilePath, configName, kClass, vClass);
+    }
+
+    private static <T> Optional<T> getValuesBySpecConfigName(Config config, Path openApiFilePath,
+            CodegenConfig.ConfigName configName,
+            Class<T> propertyType) {
+        return config
+                .getOptionalValue(CodegenConfig.getSpecConfigName(configName, openApiFilePath), propertyType)
+                .or(() -> config.getOptionalValue(CodegenConfig.getGlobalConfigName(configName), propertyType));
+    }
+
+    private static <K, V> Optional<Map<K, V>> getValuesBySpecConfigName(SmallRyeConfig config, Path openApiFilePath,
+            CodegenConfig.ConfigName configName, Class<K> kClass, Class<V> vClass) {
         return config
                 .getOptionalValues(CodegenConfig.getSpecConfigName(configName, openApiFilePath), kClass, vClass)
                 .or(() -> config.getOptionalValues(CodegenConfig.getGlobalConfigName(configName), kClass, vClass));
     }
+
+    private static <T> Optional<T> getValuesByConfigKey(Config config, String configName, Class<T> propertyType,
+            CodegenConfig.ConfigName configName1) {
+        return config
+                .getOptionalValue(configName, propertyType)
+                .or(() -> config.getOptionalValue(CodegenConfig.getGlobalConfigName(configName1), propertyType));
+    }
+
+    private static <K, V> Optional<Map<K, V>> getValuesByConfigKey(SmallRyeConfig config, CodegenConfig.ConfigName configName,
+            Class<K> kClass, Class<V> vClass, String configKey) {
+        return config
+                .getOptionalValues(CodegenConfig.getSpecConfigNameByConfigKey(configKey, configName), kClass,
+                        vClass)
+                .or(() -> config.getOptionalValues(CodegenConfig.getGlobalConfigName(configName), kClass, vClass));
+    }
+
+    private static Optional<String> getConfigKeyValue(Config config, Path openApiFilePath) {
+        String configKey = String.format("quarkus.openapi-generator.codegen.spec.%s.%s", getSanitizedFileName(openApiFilePath),
+                CONFIG_KEY_PROPERTY);
+        return config.getOptionalValue(configKey, String.class);
+    }
+
+    private static Optional<String> getConfigKeyValue(SmallRyeConfig config, Path openApiFilePath) {
+        String configKey = String.format("quarkus.openapi-generator.codegen.spec.%s.%s", getSanitizedFileName(openApiFilePath),
+                CONFIG_KEY_PROPERTY);
+        return config.getOptionalValue(configKey, String.class);
```

Comment:
Can we rethink these methods? They are too similar and hard to grasp the minimal changes between them besides the signature overload.

Reply:
It is true. `io.smallrye.config.SmallRyeConfig` implements `org.eclipse.microprofile.config`. Is not necessary to have bother methods :smile:. I removed the most specific